packages/opensrp-plans/src/ducks/index.ts

```diff
@@ -0,0 +1,352 @@
+import { Dictionary } from '@onaio/utils/dist/types/types';
+import intersect from 'fast_array_intersect';
+import { get, keyBy, values } from 'lodash';
+import { AnyAction, Store } from 'redux';
+import { createSelector } from 'reselect';
+import SeamlessImmutable from 'seamless-immutable';
+import { PlanDefinition, UseContext } from './configs/settings';
+import { descendingOrderSort, isPlanDefinitionOfType } from '../helpers/utils';
+
+/** the reducer name */
+export const reducerName = 'PlanDefinition';
+
+/** list of plan types displayed
+ * TODO: To be moved to env
+ */
+export const DISPLAYED_PLAN_TYPES = 'FI,IRS,MDA,MDA-Point,Dynamic-FI,Dynamic-IRS,Dynamic-MDA';
+
+// actions
+
+/** PLAN_DEFINITION_FETCHED action type */
+export const PLAN_DEFINITIONS_FETCHED =
+  'reveal/reducer/opensrp/PlanDefinition/PLAN_DEFINITIONS_FETCHED';
+
+/** PLAN_DEFINITION_FETCHED action type */
+export const REMOVE_PLAN_DEFINITIONS =
+  'reveal/reducer/opensrp/PlanDefinition/REMOVE_PLAN_DEFINITIONS';
+
+/** PLAN_DEFINITION_FETCHED action type */
+export const ADD_PLAN_DEFINITION = 'reveal/reducer/opensrp/PlanDefinition/ADD_PLAN_DEFINITION';
+
+/** interface for plan Objects */
+/** Enum representing the possible intervention types */
+export enum PlanStatus {
+  ACTIVE = 'active',
+  COMPLETE = 'complete',
+  DRAFT = 'draft',
+  RETIRED = 'retired',
+}
+
+/** PlanRecord - base Plan interface for plan objects,  keyed by `id` in state */
+export interface PlanRecord {
+  id: string;
+  plan_date: string;
+  plan_effective_period_end: string;
+  plan_effective_period_start: string;
+  plan_fi_reason: string;
+  plan_fi_status: string;
+  plan_id: string;
+  plan_intervention_type: InterventionType;
+  plan_status: PlanStatus;
+  plan_title: string;
+  plan_useContext?: UseContext[];
+  plan_version?: string;
+  plan_jurisdictions_ids?: string[];
+}
+
+/** Plan - interface for plan[-jurisdiction] objects */
+export interface Plan extends PlanRecord {
+  jurisdiction_depth: number;
+  jurisdiction_id: string;
+  jurisdiction_name: string;
+  jurisdiction_name_path: string[];
+  jurisdiction_parent_id: string;
+  jurisdiction_path: string[];
+}
```

Comment:
This describes the plan entity as received from superset, we will not be needing this for opensrp-web, at least not for now.